service/history/mutableStateBuilder.go

```diff
@@ -279,6 +279,9 @@ func (e *mutableStateBuilder) Load(
 	e.pendingActivityInfoIDs = state.ActivityInfos
 	for _, activityInfo := range state.ActivityInfos {
 		e.pendingActivityIDToEventID[activityInfo.ActivityId] = activityInfo.ScheduleId
+		if activityInfo.HeartbeatTimeout == nil || *activityInfo.HeartbeatTimeout > 0 {
```

Comment:
Should add the rest of the checks and refactor all the other things timersequence does as well.